### ISSUE TITLE
Editor annotating doesn't work

### ISSUE THREAD

**Issue by szabope** [szabope/mypy-pycharm-plugin#111]
> Good news: the settings in Preferences → Tools → Mypy work correctly, and the Mypy status in the toolbar also looks fine.
> 
> However, the editor area doesn’t show error underlines as expected.
> 
> <img width="1025" height="185" alt="Image" src="https://github.com/user-attachments/assets/bac67c8d-a7bf-40c2-ab4f-f384defbac96" />
> 
> <img width="286" height="174" alt="Image" src="https://github.com/user-attachments/assets/e664a6cb-ad70-4095-9d62-96bae1b61f9e" />
> 
> In some cases, it does work — I’m sure I once opened a project where the underlines appeared correctly, but I haven’t been able to reproduce it since.
> 
> I’ve also tried switching between different themes, but the issue remains, so it’s probably not theme-related.
> 
> Additionally, the “Scan” button in the toolbar needs to be clicked manually to trigger a run. I remember it used to scan automatically and show errors without manual triggering — has that behavior changed?
>  

 _Originally posted by @Nagato-Yuzuru in [#108](https://github.com/szabope/mypy-pycharm-plugin/issues/108#issuecomment-3478857797)_

**Comment by szabope** [comment]
The expected behavior is still the same, as you have experienced it once: file focused in editor is automatically checked; you can trigger a run on multiple files / directories from the right-click menu or from the toolbar....

I tried to reproduce the issue, but it works with my setup.

Let's figure out the difference.
This issue seems to be from a different project.

Please provide values for `Arguments` and `Config file` from plugin config, also the contents of `Config file` if any.

If `Config file` is empty and you have a `pyproject.toml` please share that one, too, with sensitive parts redacted, of course.

Can you please check Pycharm logs for mypy related problems? (Help > Open Log in Editor) Look for `MypyAnnotator`

**Comment by szabope** [comment]
Hm...

Monday Morning Mystery :)

I'm thinking. I'll get back to you when I have something.

Thank you for all your help.

**Comment by szabope** [comment]
> Interestingly, another file in the same project works correctly.

The issue being related to particular files gives me an idea: what if I missed something with error handling. I'll have a look today a bit later. Thanks.
I'll keep you updated.

**Comment by szabope** [comment]
Thank you for testing the new version and all your help before.
> the option to use mypy directly from the project SDK that’s very convenient!

I'm glad that you found it useful. A step towards supporting WSL.

So the missing annotations issue with the old project seems to come from mypy? Or is there still a chance that the plugin has the issue?

About mypy skipping packages: I don't have many ideas for now, but I'll let you know if something pops into my mind.

This exception is a very valueable one: I messed up something, because it should be totally impossible to happen. I'm on it.

**Comment by Nagato-Yuzuru** [comment]
Thank you for your time.

I’ve updated to version 2.0.1, but it doesn’t seem to make much difference, the issue still persists.

In the same project, if I add code like this in different files:
```python

def f(s: str):
    ...

f(1)
```

some files still fail to show the expected error, though the Scan window displays the correct result.

> expection

<img width="132" height="114" alt="Image" src="https://github.com/user-attachments/assets/95a407aa-073c-4074-aba4-7b54d7d8d657" />

<img width="1423" height="308" alt="Image" src="https://github.com/user-attachments/assets/9ae22211-09ca-4393-bcbc-524d21984618" />

>  no show wrong annotating

<img width="166" height="123" alt="Image" src="https://github.com/user-attachments/assets/654123e7-42bc-438c-b831-bdb9e9bdb88f" />

<img width="1765" height="347" alt="Image" src="https://github.com/user-attachments/assets/3a1dba84-a1ef-42bf-990a-ed38bb3cb196" />

idk maybe related to this: in the files where mypy fails to show errors, PyCharm’s own static analysis also seems to stop working (for example, the green squiggly underlines disappear).

> uninstall mypy plugin:
correct work file:

<img width="156" height="134" alt="Image" src="https://github.com/user-attachments/assets/e7663ebc-d935-4d3a-bc67-8ef068d41b05" />

no wrong annotating file:
 
<img width="157" height="134" alt="Image" src="https://github.com/user-attachments/assets/38cc285b-56b9-418e-875c-882169e499eb" />

**Comment by Nagato-Yuzuru** [comment]
Incidentally, after updating to 2.0.1, editing text becomes extremely sluggish. Disabling the plugin or reverting to 2.0.1 resolves the issue. I use ideavim.
